agent/consul/server.go

```diff
@@ -376,8 +382,8 @@ func NewServer(config *Config, flat Deps) (*Server, error) {
 		leaveCh:                 make(chan struct{}),
 		reconcileCh:             make(chan serf.Member, reconcileChSize),
 		router:                  flat.Router,
-		rpcServer:               rpc.NewServer(),
-		insecureRPCServer:       rpc.NewServer(),
+		rpcServer:               rpc.NewServerWithOpts(rpc.WithServerServiceCallInterceptor(GetNetRPCInterceptor(secureRecorder))),
+		insecureRPCServer:       rpc.NewServerWithOpts(rpc.WithServerServiceCallInterceptor(GetNetRPCInterceptor(insecureRecorder))),
```

Comment:
Project from Alex Pana to add interceptors to net/rpc. Allows us to more easily create a span for every RPC we receive as a server